cache/api/cache.api

```diff
@@ -1,238 +1,30 @@
-public abstract class com/dropbox/android/external/cache3/AbstractFuture : com/dropbox/android/external/cache3/ListenableFuture {
-	protected fun <init> ()V
-	public fun addListener (Ljava/lang/Runnable;Ljava/util/concurrent/Executor;)V
-	public fun cancel (Z)Z
-	public fun get ()Ljava/lang/Object;
-	public fun get (JLjava/util/concurrent/TimeUnit;)Ljava/lang/Object;
-	protected fun interruptTask ()V
-	public fun isCancelled ()Z
-	public fun isDone ()Z
-	protected fun set (Ljava/lang/Object;)Z
-	protected fun setException (Ljava/lang/Throwable;)Z
-	protected fun setFuture (Lcom/dropbox/android/external/cache3/ListenableFuture;)Z
-	protected final fun wasInterrupted ()Z
-}
-
-public abstract class com/dropbox/android/external/cache3/AbstractFuture$TrustedFuture : com/dropbox/android/external/cache3/AbstractFuture {
-	public fun <init> ()V
-	public final fun addListener (Ljava/lang/Runnable;Ljava/util/concurrent/Executor;)V
-	public final fun get ()Ljava/lang/Object;
-	public final fun get (JLjava/util/concurrent/TimeUnit;)Ljava/lang/Object;
-	public final fun isCancelled ()Z
-	public final fun isDone ()Z
-}
-
-public abstract class com/dropbox/android/external/cache3/AbstractSequentialIterator : com/dropbox/android/external/cache3/UnmodifiableIterator {
-	protected fun <init> (Ljava/lang/Object;)V
-	protected abstract fun computeNext (Ljava/lang/Object;)Ljava/lang/Object;
-	public final fun hasNext ()Z
-	public final fun next ()Ljava/lang/Object;
-}
-
-public final class com/dropbox/android/external/cache3/Ascii {
-	public static fun isUpperCase (C)Z
-	public static fun toLowerCase (Ljava/lang/String;)Ljava/lang/String;
-}
-
-public abstract interface class com/dropbox/android/external/cache3/Cache {
-	public abstract fun asMap ()Ljava/util/concurrent/ConcurrentMap;
-	public abstract fun cleanUp ()V
-	public abstract fun get (Ljava/lang/Object;Ljava/util/concurrent/Callable;)Ljava/lang/Object;
-	public abstract fun getAllPresent (Ljava/lang/Iterable;)Ljava/util/Map;
+public abstract interface class com/dropbox/kmp/external/cache3/Cache {
 	public abstract fun getIfPresent (Ljava/lang/Object;)Ljava/lang/Object;
+	public abstract fun getOrPut (Ljava/lang/Object;Lkotlin/jvm/functions/Function0;)Ljava/lang/Object;
```

Comment:
Curious what is the new api used for? I'm always hesitant to support more public functionality

Reply:
I introduced this because:
`public abstract fun get (Ljava/lang/Object;Ljava/util/concurrent/Callable;)Ljava/lang/Object;`
is not implemented yet.